armi/runLog.py

```diff
@@ -26,188 +26,157 @@
     runLog.error('extra error info here')
     raise SomeException  # runLog.error() implies that the code will crash!
 
-.. note::
-    We plan to reimplement this with the standard Python logging module. It was customized
-    to add a few features in a HPC/MPI environment but we now think we can use the standard
-    system.
+You can interact with the logger in much the same way now by doing:
+
+.. code::
+
+    import logging
+    runLog = logging.getLogger('whatever')
 
 """
 from __future__ import print_function
-import sys
-import os
+from glob import glob
 import collections
+import logging
 import operator
+import os
+import sys
 import time
 
-from six import StringIO
-
 from armi import context
-from armi.context import Mode
-from armi import meta
-
-
-# use ordereddict so we can get right order of options in GUI.
-_logLevels = collections.OrderedDict(
-    [
-        ("debug", (0, "dbug")),
-        ("extra", (10, "xtra")),
-        ("info", (20, "info")),
-        ("important", (25, "impt")),
-        ("prompt", (27, "prmt")),
-        ("warning", (30, "warn")),
-        ("error", (50, "err ")),
-        ("header", (100, "")),
-    ]
-)
-
-_stderrName = "{0}.{1:04d}.stderr"
-_stdoutName = "{0}.{1:04d}.stdout"
-
-
-def getLogVerbosityLevels():
-    """Return a list of the available log levels (e.g., debug, extra, etc.)."""
-    return list(_logLevels.keys())
-
-
-def getLogVerbosityRank(level):
-    """Return integer verbosity rank given the string verbosity name."""
-    if level not in getLogVerbosityLevels():
-        raise KeyError(
-            "{} is not a valid verbosity level. Choose from {}".format(
-                level, getLogVerbosityLevels()
-            )
-        )
-    return _logLevels[level][0]
-
-
-def _checkLogVerbsityRank(rank):
-    """Check that the verbosity rank is defined within the _logLevels and return it if it is."""
-    validRanks = []
-    for level in getLogVerbosityLevels():
-        expectedRank = getLogVerbosityRank(level)
-        validRanks.append(rank)
-        if rank == expectedRank:
-            return rank
-    raise KeyError(
-        "Invalid verbosity rank {}. Valid options are: {}".format(rank, validRanks)
-    )
 
 
-class Log:
+# global constants
+_SINGLE_STR = "|SINGLE|"
+_WHITE_SPACE = " " * 6
+_ADD_LOG_METHOD_STR = """def {0}(self, message, *args, **kws):
+    if self.isEnabledFor({1}):
+        self._log({1}, message, args, **kws)
+logging.Logger.{0} = {0}"""
+
+
+class RunLog:
```

Comment:
So this class is mostly responsible for standing up and tearing down the logging stuff. It's used by the module-level logging functions (old API), but modules that wish to opt into the `logging.getLogger()` approach (new API) seem not to need any special knowledge of the `RunLog` class. Is this accurate? Should this be private?

Reply:
That's an interesting point.  Yeah, we could make that class private.  The only place it is used is in `armi/test/mockRunLogs.py`, which subclasses `RunLog` for testing purposes. 

I guess it depends on how hard-line you want to be about following the "don't use a private class" rule in `mockRunLogs.py`.  If you don't mind importing the private class there, we can do it.  Your call.